Classes/Common/IiifManifest.php

```diff
@@ -126,24 +127,35 @@ protected function establishRecordId($pid)
              *  check the configuration.
              *  TODO Saving / indexing should still work - check!
              */
-            $result = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
-                'tx_dlf_metadataformat.xpath AS querypath',
-                'tx_dlf_metadata,tx_dlf_metadataformat,tx_dlf_formats',
-                'tx_dlf_metadata.pid=' . intval($pid)
-                    . ' AND tx_dlf_metadataformat.pid=' . intval($pid)
-                    . ' AND ((tx_dlf_metadata.uid=tx_dlf_metadataformat.parent_id AND tx_dlf_metadataformat.encoded=tx_dlf_formats.uid'
-                    . ' AND tx_dlf_metadata.index_name="record_id" AND tx_dlf_formats.type=' . $GLOBALS['TYPO3_DB']->fullQuoteStr($this->getIiifVersion(), 'tx_dlf_formats') . ') OR tx_dlf_metadata.format=0)'
-                    . Helper::whereClause('tx_dlf_metadata', true)
-                    . Helper::whereClause('tx_dlf_metadataformat')
-                    . Helper::whereClause('tx_dlf_formats')
-            );
-            if ($GLOBALS['TYPO3_DB']->sql_num_rows($result) > 0) {
-                for ($i = 0, $j = $GLOBALS['TYPO3_DB']->sql_num_rows($result); $i < $j; $i++) {
-                    $resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result);
-                    $recordIdPath = $resArray['querypath'];
-                    if (!empty($recordIdPath)) {
-                        $this->recordId = $this->iiif->jsonPath($recordIdPath);
-                    }
+            $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
+                ->getQueryBuilderForTable('tx_dlf_metadata');
+            // Get hidden records, too.
+            $queryBuilder
+                ->getRestrictions()
+                ->removeByType(HiddenRestriction::class);
+            $result = $queryBuilder
+                ->select('tx_dlf_metadataformat.xpath AS querypath')
+                ->from('tx_dlf_metadata')
+                ->from('tx_dlf_metadataformat')
+                ->from('tx_dlf_formats')
+                ->where(
+                    $queryBuilder->expr()->eq('tx_dlf_metadata.pid', intval($pid)),
+                    $queryBuilder->expr()->eq('tx_dlf_metadataformat.pid', intval($pid)),
+                    $queryBuilder->expr()->orX(
```

Comment:
Can't use a proper `join()` (or `innerJoin()` as in `MetsDocument`) because of this `OR` statement...